src/exp_and_gram.jl

```diff
@@ -248,20 +299,23 @@ function _exp_and_gram_chol_init!(
         end
     end
 
-    odd = A * odd
-    den = even - odd # pade denominator
-    num = even + odd # pade numerator
+    odd = mul!(tmpA1, A, odd)
+    den = tmpA2 .= even .- odd # pade denominator
+    num = tmpA3 .= even .+ odd # pade numerator
 
-    Lodd .= A * Lodd
+    Lodd .= mul!(Loddtmp, A, Lodd) # writes into L as Lodd and L share memory
 
     F = lu!(den)
     expA = num
     ldiv!(F, expA)
     ldiv!(F, L)
 
-    U = qr!(L').R # right Cholesky factor of the Grammian (may not be square!!)
-    U = triu2cholesky_factor!(U)
-    return expA, U
+    _U = qr!(L').R # right Cholesky factor of the Grammian (may not be square!!)
+    _U = triu2cholesky_factor!(_U)
+
+    copy!(eA, expA)
+    copy!(U, _U)
```

Comment:
I think this is a case of incomplete tests again. ```U``` is always square but as it is implemented now, ```_U = qr!(L').R``` need not be square.

Reply:
So that means the tests introduced in https://github.com/filtron/FiniteHorizonGramians.jl/pull/17/files would fail here, right? I could rebase after the merge and figure out how to get them to pass.

Reply:
Or if you have a concrete code suggestion here please let me know

Reply:
Ideally the added tests would make this fail. Another alternative is to write the non-square R factor into e.g. U and then setting the unused elements in U to zero. If the initial factor is square the doubling method could be simplified as well.

Reply:
So if I understand correctly, previously you actually returned a non-square square-root, but now that we assume that `U` is square, so we just need to fill it with `_U` such that `_U'_U == U'U`? So this here should fix it I guess:
```julia
    U .= 0
    copy!(view(U, 1:size(_U, 1), 1:size(_U, 2)), _U)
```
Not sure how performant or not this is. But tests pass (which they didn't before the change, after the rebase!)

Reply:
(I committed and pushed the changes)

Reply:
I think the implementation looks correct now!